server/command.go

```diff
@@ -121,6 +125,90 @@ func (p *Plugin) ExecuteCommand(c *plugin.Context, args *model.CommandArgs) (*mo
 		}
 		p.postCommandResponse(args, str.String())
 		return &model.CommandResponse{}, nil
+	case "set_channel_welcome":
```

Comment:
Would you mind refactoring each `case` block into a function call?  For an example, you can view here https://github.com/mattermost/mattermost-plugin-agenda/blob/pluginReview/server/command.go#L55

This has several benefits:
* The switch statement slash command handlers are easier to read.
* Each refactored `execute` function will not need to return a `nil` `model.AppError`
  Each case in `ExecuteCommand` will do the nil return
  Example: `return p.executeCommandList(args), nil`

I noticed you already mentioned this in your PR description, after I made this comment, but you now have another plugin example :)

Reply:
This is looking great, @vespian! let's do a round of PR changes and re-review!  

Regarding the ephemeral posts not working.  Instead of using `p.postCommandResonse()` and returning an empty `model.CommandResponse{}` object, you can return the following `commandResponse` to post an ephemeral message.

```go
return &model.CommandResponse{
	ResponseType: model.COMMAND_RESPONSE_TYPE_EPHEMERAL,
	Text:         "HELLO",
	Type:         model.POST_DEFAULT,
}, nil
// p.postCommandResponse(args, string(data))
```

We've used this helper function in a couple of plugins to help mitigate having this return statement all over the code.  
https://github.com/mattermost/mattermost-plugin-agenda/blob/pluginReview/server/command.go#L177

Reply:
@jfrerich Maybe we keep the code as it as for now and DRY it up in another PR? 0/5

Reply:
> Regarding the ephemeral posts not working. Instead of using p.postCommandResonse() and returning an empty model.CommandResponse{} object, you can return the following commandResponse to post an ephemeral message.

@jfrerich This helps to DRY the code in `command.go`, but the original issue is that I somehow can't send an ephemeral message from the `UserHasJoinedChannel` hook, only normal posts:

https://github.com/vespian/mattermost-plugin-welcomebot/blob/5db63504cbd9b41a320d49def24baf685a7e0862/server/hooks.go#L57-L67

And it results in the welcome messages being visible to everybody in the channel (i.e. normal posts) and it is not optimal. Do you know maybe how we can improve that?

Reply:
> @jfrerich Maybe we keep the code as it as for now and DRY it up in another PR? 0/5

Normally I would +1 you, but the PR review cycle is pretty long. I will prepare the change requested by @jfrerich as a separete PR.

Reply:
> Would you mind refactoring each case block into a function call?

@jfrerich Done.